ui/js/controllers/perf/e10s-trend.js

```diff
@@ -0,0 +1,568 @@
+                            }
+                        }
+                    }
+                });
+            });
+
+            $scope.testList = Object.keys($scope.compareResults);
+            console.log('compareResults:');
+            console.log($scope.compareResults);
+        }
+
+        function loadData() {
+            var resultsMap = {
+                e10s: {},
+                base: {}
+            };
+            $scope.baseTestList = [];
+            $scope.newTestList = [];
+            $scope.dataLoading = true;
+            $scope.baseResults = {};
+            $scope.newResults = {};
+            $scope.titles = {};
+
+            // ** get dates for original base date (for the last x days back from the base date) **
+            // selectedBaseDate and selectedNewDate are in SECONDS from today back i.e. 1 day ago, 1 week ago
+
+            // calculate the actual base date itself, as a date not just x seconds back from today
+            var today_epoch = Date.now();
+            var base_date = today_epoch - ($scope.selectedBaseDate.value * 1000)  // want MS
+
+            // start date for retrieving data is the base date minus sample size i.e. x days back from base date
+            var data_start_date = base_date - ($scope.selectedSampleSize.value * 1000)  // want MS
+            data_start_date = new Date(data_start_date);
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            $scope.base_data_start_date = data_start_date.toISOString().slice(0, -5);
+
+            // end date for retrieving data is the base date itself
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            var data_end_date =  new Date(base_date);
+            $scope.base_data_end_date = data_end_date.toISOString().slice(0, -5);
+
+            // ** get dates for new date (for the last x days back from the new date) **
+            // calculate the actual new date itself, as a date not just x seconds back from today
+            var today_epoch = Date.now();
+            var new_date = today_epoch - ($scope.selectedNewDate.value * 1000)  // want MS
+
+            // start date for retrieving data is the new date minus sample size i.e. x days back from new date
+            var data_start_date = new_date - ($scope.selectedSampleSize.value * 1000)  // want MS
+            data_start_date = new Date(data_start_date);
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            $scope.new_data_start_date = data_start_date.toISOString().slice(0, -5);
+
+            // end date for retrieving data is the new date itself
+            // must be in format 'YYYY-MM-DDT:HH:MM:SS' ie. 2016-09-10T1:1:1
+            var data_end_date =  new Date(new_date);
+            $scope.new_data_end_date = data_end_date.toISOString().slice(0, -5);
```

Comment:
A couple things:

* We use camel case for javascript (i.e. $scope.newDataEndData, not $scope.new_data_end_date)
* It's only necessary to set a $scope property when you want to access it from an angular template (which it doesn't look like you want to here).
* This is pretty verbose :) How about something like this:

```js
function getDateRange(fromDateDelta, sampleSize) {
   var fromDateMs = Date.now() - (fromDateDelta * 1000);
   return {
      start: new Date(fromDateMs - sampleSize).toISOString().slice(0, -5),
      end: new Date(fromDateMs).toISOString().slice(0, -5)
  }
}

var baseDateRange = getDateRange($scope.selectedBaseDate, $scope.selectedSampleSize);
var newDateRange = getDateRange($scope.selectedNewDate, $scope.selectedSampleSize);
```